pkg/operator/starter.go

```diff
@@ -547,15 +550,38 @@ func prepareOauthAPIServerOperator(ctx context.Context, controllerContext *contr
 	).WithStaticResourcesController(
 		"APIServerStaticResources",
 		assets.Asset,
-		[]string{
-			"oauth-apiserver/ns.yaml",
-			"oauth-apiserver/apiserver-clusterrolebinding.yaml",
-			"oauth-apiserver/svc.yaml",
-			"oauth-apiserver/sa.yaml",
-			"oauth-apiserver/RBAC/useroauthaccesstokens_binding.yaml",
-			"oauth-apiserver/RBAC/useroauthaccesstokens_clusterrole.yaml",
-			"oauth-apiserver/oauth-apiserver-pdb.yaml",
+		[]apiservercontrollerset.ConditionalFiles{
+			{
+				Files: []string{
+					"oauth-apiserver/ns.yaml",
+					"oauth-apiserver/apiserver-clusterrolebinding.yaml",
+					"oauth-apiserver/svc.yaml",
+					"oauth-apiserver/sa.yaml",
+					"oauth-apiserver/RBAC/useroauthaccesstokens_binding.yaml",
+					"oauth-apiserver/RBAC/useroauthaccesstokens_clusterrole.yaml",
+				},
+			},
+			{
+				Files: []string{
+					"oauth-apiserver/oauth-apiserver-pdb.yaml",
+				},
+				ShouldCreateFn: func() bool {
+					isSno, err := guard.IsSNOCheckFnc(infraLister)()
+					if err != nil {
+						return false
+					}
+					return !isSno
+				},
+				ShouldDeleteFn: func() bool {
+					isSno, err := guard.IsSNOCheckFnc(infraLister)()
+					if err != nil {
```

Comment:
I think you'll have a nicer time if you make the `IsSNOCheckFnc` only return a bool